modules/c++/sys/source/OSUnix.cpp

```diff
@@ -346,6 +373,39 @@ void sys::OSUnix::getMemInfo(size_t &totalPhysMem, size_t &freePhysMem) const
 #endif
 }
 
+std::string sys::OSUnix::getCurrentExecutable(
+        const std::string& argvPathname) const
+{
+    std::vector<std::string> possibleSymlinks;
+
+    // Linux
+    possibleSymlinks.push_back(sys::Path::joinPaths(
+            sys::Path::delimiter()[0] + std::string("proc"),
+            sys::Path::joinPaths("self", "exe")));
+
+    // Solaris
+    possibleSymlinks.push_back(sys::Path::joinPaths(
+            sys::Path::delimiter()[0] + std::string("proc"),
+            sys::Path::joinPaths("self",
+            sys::Path::joinPaths("path", "a.out"))));
+
+    for (size_t ii = 0; ii < possibleSymlinks.size(); ++ii)
+    {
+        struct stat linkStatus;
+        const std::string pathname = possibleSymlinks[ii];
+        linkStat(pathname, linkStatus);
```

Comment:
So actually... why is `linkStat()` called?  Nothing does anything with `linkStatus`.  Did a check or some other usage get removed in some of the refactoring?